torch/csrc/jit/codegen/cuda/lower_unroll.cpp

```diff
@@ -124,40 +124,43 @@ kir::Expr* findVectorizedSet(
 
 // Get full extent for the inner-most, merged root domain
 kir::Val* getVectorizeExtent(
-    kir::TensorView* tv,
-    const std::vector<kir::Val*>& indices) {
-  auto domain = tv->domain()->hasRFactor() ? tv->domain()->rfactorDomain()
-                                           : tv->domain()->rootDomain();
+    kir::TensorView* global_tv,
+    kir::TensorView* local_tv) {
+  TORCH_INTERNAL_ASSERT(!global_tv->domain()->hasRFactor());
+  auto root_domain = global_tv->domain()->rootDomain();
 
-  TORCH_INTERNAL_ASSERT(domain.size() == indices.size());
-
-  bool is_contiguous = true;
-  for (auto status : tv->domain()->contiguity()) {
-    is_contiguous &= status;
-  }
-
-  // If the tensorview is not contiguous, return inner-most root domain extent
-  if (!is_contiguous) {
-    return domain.back()->extent();
-  }
+  auto local_fuser_tv = local_tv->fuserTv();
+  auto root_right_of_ca_domains = IterVisitor::getInputsTo(
+      {local_fuser_tv->domain()->domain().begin() +
+           local_fuser_tv->getComputeAtPosition(),
+       local_fuser_tv->domain()->domain().end()});
 
   const auto gpu_lower = GpuLower::current();
   kir::IrBuilder ir_builder(gpu_lower->kernel());
 
   // Calculate extent of merged root domains
   kir::Val* extent = nullptr;
-  for (int i = int(domain.size()) - 1; i >= 0; --i) {
-    auto root_id = domain.at(i);
+  for (int i = int(root_domain.size()) - 1; i >= 0; --i) {
+    auto root_id = root_domain.at(i);
     if (root_id->isBroadcast() || root_id->isReduction() ||
         gpu_lower->trivialReductionInfo().isDerived(root_id)) {
       continue;
-    } else if (extent == nullptr) {
+    }
+    // Don't extend the vectorization domain beyond the CA position
+    if (root_right_of_ca_domains.find(local_fuser_tv->getRootDomain()[i]) ==
```

Comment:
Would a `PairwiseRootDomainMap` make sense here to ensure a correct mapping between `global_tv` and `local_tv`?

Reply:
That's a good point. I'll fix it.